electrum/three_keys/script.py

```diff
@@ -0,0 +1,106 @@
+from typing import List
+
+from electrum.bitcoin import opcodes, push_script
+from electrum.three_keys.multisig_generator import MultisigScriptGenerator
+
+
+class ThreeKeysError(Exception):
+    pass
+
+
+class TwoKeysScriptGenerator(MultisigScriptGenerator):
+    def __init__(self, recovery_pubkey: str):
+        self.recovery_pubkey = recovery_pubkey
+        self._recovery_alert_flag = None
+
+    def get_redeem_script(self, public_keys: List[str]) -> str:
+        if not isinstance(public_keys, list) or len(public_keys) != 1:
+            raise ThreeKeysError(f"Wrong input type! Expected list not '{public_keys}'")
+
+        pub_key = public_keys[0]
+        return (
+                opcodes.OP_IF.hex() +
+                opcodes.OP_1.hex() +
+                opcodes.OP_ELSE.hex() +
+                opcodes.OP_2.hex() +
+                opcodes.OP_ENDIF.hex() +
+
+                push_script(pub_key) +
+                push_script(self.recovery_pubkey) +
+
+                opcodes.OP_2.hex() +
+                opcodes.OP_CHECKMULTISIG.hex()
+        )
+
+    def get_script_sig(self, signatures: List[str], public_keys: List[str]) -> str:
+        if self._recovery_alert_flag is None:
+            raise ThreeKeysError('Recovery/alert flag not set!')
+        sigs = ''.join(push_script(sig) for sig in signatures)
+        return (
+            opcodes.OP_0.hex() +
+            sigs +
+            self._recovery_alert_flag +
+            push_script(self.get_redeem_script(public_keys))
+        )
+
+    def set_alert(self):
+        # 1 of 2
+        self._recovery_alert_flag = opcodes.OP_1.hex()
```

Comment:
Please consider this:
According to https://github.com/bitcoinvault/bitcoinvault/blob/master/src/script/interpreter.h#L105 segwit compatible OP_IFs should get `minimal` form of arguments so:
- use `0x01` for OP_TRUE
- empty vector for OP_FALSE
It is already implemented in bitcoinvault:
https://github.com/bitcoinvault/bitcoinvault/blob/dev/src/script/sign.cpp#L157

This also applies to lines: 52, 98, 102, 106